### PR TITLE
task/polygons/send_quadrilateral_action

### PR DESCRIPTION
**Pull request by BarAgam**

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    public void onShowButtonClicked() {
+        if (!isFormFilled()) {
+            mView.showInvalidInput();
+        } else {
+            drawNewPolygon();
+            centerMapOnPolygon();
+        }
+    }
+
+    private void centerMapOnPolygon() {
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java]: Not in order

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            if (!picker.hasPoint()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void sendPolygon() {
+        mSendGeoMessageInteractorFactory
+                .create(EMTPY_STRING, createPolygon(), EMTPY_STRING)
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java]: We can offer the user to send text with the polygon...

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        }
+    }
+
+    private void drawPolygon() {
+        mPolygonEntity = createPolygonEntity();
+        mGGMapView.updateMapEntity(GeoEntityNotification.createAdd(mPolygonEntity));
+    }
+
+    private PolygonEntity createPolygonEntity() {
+        return new PolygonEntity(EMTPY_STRING, EMTPY_STRING, createPolygon(), createSymbol());
```
Comment (teamagam5) [inline app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java]: same comment...

File: app/src/main/res/layout/fragment_send_quadrilateral.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/send_quadrilateral_action_description" />
+
+
+        <com.teamagam.gimelgimel.app.common.base.view.HorizontalLongLatPicker
+            android:id="@+id/send_quadrilateral_action_long_lat1"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            horizontallonglatpicker:label="Point 1" />
```
Comment (teamagam5) [inline app/src/main/res/layout/fragment_send_quadrilateral.xml]: extract resource?

File: app/src/main/res/layout/horizontal_long_lat_picker.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    android:orientation="horizontal">
+
+    <me.grantland.widget.AutofitTextView
+        android:id="@+id/horizontal_long_lat_picker_text_view"
+        android:layout_width="0dp"
+        android:layout_height="match_parent"
+        android:layout_weight="1"
+        android:gravity="center_vertical"
+        android:maxLines="1"
+        android:textSize="30sp"
```
Comment (teamagam5) [inline app/src/main/res/layout/horizontal_long_lat_picker.xml]: extract resource?

**Review by teamagam5 — APPROVED** [review]
general: extract resources if possible in xmls

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java]: K

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java (hunk 2 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java]: Ehumm I'll add that

File: app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java (hunk 3 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendQuadrilateralActionViewModel.java]: ^^

File: app/src/main/res/layout/fragment_send_quadrilateral.xml (hunk 4 above)
Comment (BarAgam) [reply app/src/main/res/layout/fragment_send_quadrilateral.xml]: K

File: app/src/main/res/layout/horizontal_long_lat_picker.xml (hunk 5 above)
Comment (BarAgam) [reply app/src/main/res/layout/horizontal_long_lat_picker.xml]: K

**Pull request merged**